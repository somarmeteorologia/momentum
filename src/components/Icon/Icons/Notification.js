import React from 'react'

export default function Notification({ width, height, color }) {
  return (
    <svg width={width} height={height} viewBox="0 0 32 30" fill="none">
      <path
        d="M24.8 15V10.8C24.8 8.20088 23.7675 5.70821 21.9296 3.87035C20.0918 2.0325 17.5991 1 15 1C12.4009 1 9.90821 2.0325 8.07035 3.87035C6.2325 5.70821 5.2 8.20088 5.2 10.8V15C5.2 19.62 1 20.74 1 23.4C1 25.78 6.46 27.6 15 27.6C23.54 27.6 29 25.78 29 23.4C29 20.74 24.8 19.62 24.8 15Z"
        stroke={color}
        strokeWidth="1.4"
        strokeMiterlimit="10"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M15.0004 30.3999C13.585 30.3999 12.255 30.3523 11.0034 30.2599C11.2702 31.1111 11.8018 31.8548 12.5208 32.3829C13.2397 32.9109 14.1084 33.1957 15.0004 33.1957C15.8924 33.1957 16.7612 32.9109 17.4801 32.3829C18.199 31.8548 18.7306 31.1111 18.9974 30.2599C17.7458 30.3523 16.4158 30.3999 15.0004 30.3999Z"
        fill={color}
      />
    </svg>
  )
}
