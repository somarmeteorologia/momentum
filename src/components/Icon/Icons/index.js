export * from './Add'
export * from './Advance'
export * from './Agriculture'
export * from './Alert'
export * from './Arrow'
export * from './ArrowRight'
export * from './Back'
export * from './BarChart'
export * from './Calendar'
export * from './Ch'
export * from './Chat'
export * from './Chart'
export * from './Checked'
export * from './Checklist'
export * from './City'
export * from './Clock'
export * from './Close'
export * from './Cloud'
export * from './Comment'
export * from './Configurations'
export * from './DayCC'
export * from './DayCV'
export * from './DayNB'
export * from './DayNC'
export * from './DayPC'
export * from './DayPI'
export * from './DayPI'
export * from './DayPN'
export * from './DayTR'
export * from './Disabled'
export * from './Down'
export * from './Download'
export * from './Ecomp'
export * from './Emed'
export * from './Emop'
export * from './En'
export * from './Energy'
export * from './Eye'
export * from './Favorite'
export * from './Ge'
export * from './Geo'
export * from './Home'
export * from './Info'
export * from './Layer'
export * from './Left'
export * from './Less'
export * from './Lightning'
export * from './LightningCircle'
export * from './Loading'
export * from './Location'
export * from './Manual'
export * from './Monitoring'
export * from './More'
export * from './Newsletter'
export * from './NightCC'
export * from './NightCV'
export * from './NightNB'
export * from './NightNC'
export * from './NightPC'
export * from './NightPI'
export * from './NightPN'
export * from './NightTR'
export * from './Notification'
export * from './Nv'
export * from './Options'
export * from './Output'
export * from './Pause'
export * from './Pe'
export * from './Play'
export * from './Plus'
export * from './Prevision'
export * from './Rain'
export * from './Reservoirs'
export * from './Resume'
export * from './Ret'
export * from './Right'
export * from './Segment'
export * from './Temperature'
export * from './Tv'
export * from './Up'
export * from './Valv'
export * from './Map'
export * from './Person'