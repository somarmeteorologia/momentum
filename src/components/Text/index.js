import React, { memo } from 'react'
import PropTypes from 'prop-types'

import { tokens } from '@components/Tokens'
import { Box } from '@components/Box'

import composeable from './Composeable'

import { Heading, defaultProps as headingProps } from './Heading'
import { P, defaultProps as props } from './P'

const { font } = tokens

export const Text = memo(({ children, ...props }) => {
  const Composeable = composeable('span')

  return (
    <Box {...props}>
      <Composeable {...props}>{children}</Composeable>
    </Box>
  )
})

Text.Heading = Heading
Text.P = P

const defaultProps = ({ ...props }) => ({
  ...props,
  display: 'block',
  family: 'inter'
})

Text.defaultProps = defaultProps(props)
Text.P.defaultProps = defaultProps(props)
Text.Heading.defaultProps = defaultProps(headingProps)

const defaultTypes = ({ ...types }) => ({
  ...types,
  display: PropTypes.string,
  color: PropTypes.string,
  family: PropTypes.string,
  weight: PropTypes.string
})

Text.P.propTypes = defaultTypes({
  size: PropTypes.oneOf(Object.keys(font.size).slice(5))
})
Text.Heading.propTypes = defaultTypes({
  size: PropTypes.oneOf(Object.keys(font.size).slice(0, 6))
})

const families = Object.assign(
  {},
  ...Object.keys(font.family).map(family => ({
    [family]: family
  }))
)

Text.family = families
Text.P.family = families
Text.Heading.family = families

Text.propTypes = defaultTypes({
  left: PropTypes.number,
  right: PropTypes.number,
  top: PropTypes.number,
  bottom: PropTypes.number,
  size: PropTypes.oneOf(Object.keys(font.size))
})

Text.P.size = Object.assign(
  {},
  ...Object.keys(font.size)
    .slice(5)
    .map(size => ({
      [size]: size
    }))
)

Text.size = Object.assign(
  {},
  ...Object.keys(font.size).map(size => ({
    [size]: size
  }))
)

Text.Heading.size = Object.assign(
  {},
  ...Object.keys(font.size)
    .slice(0, 6)
    .map(size => ({
      [size]: size
    }))
)

Text.weight = Object.assign(
  {},
  ...Object.keys(font.weight).map(weight => ({
    [weight]: weight
  }))
)
