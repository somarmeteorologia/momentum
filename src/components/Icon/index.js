import React, { memo, useContext } from 'react'
import PropTypes from 'prop-types'
import styled, { ThemeContext, css } from 'styled-components'
import { ifProp } from 'styled-tools'

import { Box } from '@components/Box'

import * as Icons from './Icons'

const capitalize = string => string.charAt(0).toUpperCase() + string.slice(1)

const Container = styled(Box)`
  display: flex;
  justify-content: center;
  align-items: center;
  line-height: 0;

  ${ifProp(
    'onClick',
    css`
      cursor: pointer;
    `
  )}
`

export const Icon = memo(({ name, color, ...props }) => {
  const { icons } = useContext(ThemeContext)
  const Iconable = Icons[capitalize(name)]

  return (
    <Container {...props}>
      <Iconable {...props} color={color || icons.primary} />
    </Container>
  )
})

const names = [
  'disabled',
  'city',
  'location',
  'home',
  'pause',
  'play',
  'back',
  'advance',
  'barChart',
  'lightningCircle',
  'newsletter',
  'cloud',
  'checklist',
  'checked',
  'lightning',
  'left',
  'right',
  'up',
  'down',
  'add',
  'agriculture',
  'alert',
  'arrow',
  'chart',
  'chat',
  'close',
  'comment',
  'configurations',
  'download',
  'energy',
  'favorite',
  'monitoring',
  'notification',
  'options',
  'output',
  'prevision',
  'rain',
  'reservoirs',
  'resume',
  'temperature',
  'loading',
  'plus',
  'tv',
  'clock',
  'arrowRight',
  'calendar',
  'less',
  'more',
  'layer',
  'eye',
  'info',
  'manual',
  'pe',
  'valv',
  'ret',
  'geo',
  'ecomp',
  'emed',
  'emop',
  'segment',

  'dayCC',
  'nightCC',

  'dayNB',
  'nightNB',

  'dayPN',
  'nightPN',

  'dayNC',
  'nightNC',

  'en',

  'dayPI',
  'nightPI',

  'dayPC',
  'nightPC',

  'ch',

  'ge',

  'nv',

  'dayCV',
  'nightCV',

  'dayTR',
  'nightTR'
]

Icon.propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  color: PropTypes.string,
  onClick: PropTypes.func,
  className: PropTypes.string,
  name: PropTypes.oneOf(names).isRequired
}

Icon.defaultProps = {
  width: 20,
  height: 20
}

Icon.name = Object.assign(
  {},
  ...names.map(name => ({
    [name]: name
  }))
)
