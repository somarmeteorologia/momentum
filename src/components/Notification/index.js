import React, { memo, useContext, useEffect } from 'react'
import styled, { css, ThemeContext } from 'styled-components'
import PropTypes from 'prop-types'
import posed from 'react-pose'
import { switchProp, prop } from 'styled-tools'

import { Icon } from '@components/Icon'

const Container = posed(styled.div`
  width: 385px;
  height: 110px;
  padding: 16px;
  border-radius: ${prop('theme.border.radius.four')};
  background-color: ${prop('theme.notification.bg.primary')};
  color: ${prop('theme.notification.text.primary')};
  z-index: ${prop('theme.zindex.above')};
  position: absolute;
  display: flex;
  justify-content: flex-start;
  align-items: center;

  ${switchProp('position', {
    topLeft: css`
      top: 25px;
      left: 25px;
    `,
    topRight: css`
      top: 25px;
      right: 25px;
    `,
    topCenter: css`
      top: 25px;
      left: 0;
      right: 0;
      margin-left: auto;
      margin-right: auto;
    `,
    bottomLeft: css`
      bottom: 25px;
      left: 25px;
    `,
    bottomRight: css`
      bottom: 25px;
      right: 25px;
    `,
    bottomCenter: css`
      bottom: 25px;
      left: 0;
      right: 0;
      margin-left: auto;
      margin-right: auto;
    `
  })}

  .close {
    position: absolute;
    top: 16px;
    right: 16px;
    cursor: pointer;
  }
`)({
  hidden: {
    applyAtEnd: {
      display: 'none'
    },
    opacity: 0,
    y: ({ position }) => (position.startsWith('top') ? -150 : 150)
  },
  visible: {
    applyAtStart: {
      display: 'flex'
    },
    opacity: 1,
    y: 0
  }
})

const State = styled.div`
  width: 45px;
  min-width: 45px;
  height: 45px;
  min-height: 45px;
  border-radius: ${prop('theme.border.radius.four')};
  display: flex;
  justify-content: center;
  align-items: center;

  ${switchProp('state', {
    primary: css`
      background-color: ${prop('theme.notification.state.primary')};
    `,
    danger: css`
      background-color: ${prop('theme.notification.state.danger')};
    `,
    warning: css`
      background-color: ${prop('theme.notification.state.warning')};
    `
  })};
`

const Content = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 80px;
  padding-left: 16px;
  padding-right: 16px;
`

const Title = styled.div`
  margin-bottom: 5px;
  color: ${prop('theme.colors.red.fifty')};
`

export const Notification = memo(
  ({
    title,
    description,
    isOpen,
    toClose,
    icon,
    state,
    position,
    duration
  }) => {
    const { notification } = useContext(ThemeContext)

    useEffect(() => {
      if (isOpen) {
        if (duration) {
          const timeout = setTimeout(() => {
            toClose()
          }, duration * 1000)

          return () => {
            clearTimeout(timeout)
          }
        }
      }
    }, [isOpen])

    const chooseColor = state => {
      const values = {
        primary: notification.icon.primary,
        danger: notification.icon.primary,
        warning: notification.icon.secondary
      }

      return values[state]
    }

    return (
      <Container
        initialPose="hidden"
        pose={isOpen ? 'visible' : 'hidden'}
        position={position}
      >
        <Icon
          name="close"
          className="close"
          onClick={toClose}
          width={14}
          height={14}
          color={notification.close.primary}
        />

        <State state={state}>
          {icon({
            color: chooseColor(state)
          })}
        </State>

        <Content>
          <Title>{title({ color: notification.text.primary })}</Title>
          {description({ color: notification.text.primary })}
        </Content>
      </Container>
    )
  }
)

Notification.defaultProps = {
  isOpen: false,
  state: 'primary',
  position: 'topRight',
  duration: 0,
  toClose: () => {}
}

Notification.propTypes = {
  duration: PropTypes.number,
  toClose: PropTypes.func,
  state: PropTypes.oneOf(['primary', 'danger', 'warning']),
  title: PropTypes.func.isRequired,
  description: PropTypes.func.isRequired,
  icon: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired
}

Notification.state = {
  primary: 'primary',
  danger: 'danger',
  warning: 'warning'
}

Notification.position = {
  topLeft: 'topLeft',
  topRight: 'topRight',
  topCenter: 'topCenter',
  bottomLeft: 'bottomLeft',
  bottomRight: 'bottomRight',
  bottomCenter: 'bottomCenter'
}
