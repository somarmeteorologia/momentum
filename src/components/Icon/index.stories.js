import React, { Fragment } from 'react'
import styled from 'styled-components'
import { prop } from 'styled-tools'
import { storiesOf } from '@storybook/react'

import { GROUPS } from '@helpers/hierarchySeparators'
import { Container } from '@helpers/components'

import { Reset } from '@components/Reset'
import { Icon } from '@components/Icon'

const Content = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 135px;
  height: 120px;
`

const Name = styled.span`
  margin-top: 10px;
  font-size: 14px;
  color: ${prop('theme.icons.primary')};
  font-family: ${prop('theme.font.family.inter')};
`

const Containerable = styled(Container)`
  flex-wrap: wrap;
  justify-content: flex-start;
  height: 100%;
`

const outlines = [
  'disabled',
  'home',
  'pause',
  'play',
  'back',
  'advance',
  'city',
  'barChart',
  'lightningCircle',
  'newsletter',
  'cloud',
  'checklist',
  'checked',
  'lightning',
  'left',
  'right',
  'up',
  'down',
  'add',
  'agriculture',
  'alert',
  'arrow',
  'chart',
  'close',
  'comment',
  'configurations',
  'download',
  'energy',
  'favorite',
  'monitoring',
  'notification',
  'options',
  'output',
  'prevision',
  'rain',
  'reservoirs',
  'resume',
  'temperature',
  'tv',
  'loading',
  'plus',
  'arrowRight',
  'calendar',
  'less',
  'more',
  'layer',
  'eye',
  'info',
  'manual',
  'pe',
  'valv',
  'ret',
  'geo',
  'ecomp',
  'emed',
  'emop',
  'segment'
]

const temperaturies = [
  'dayCC',
  'nightCC',

  'dayPN',
  'nightPN',

  'dayNC',
  'nightNC',

  'dayNB',
  'nightNB',

  'en',

  'dayPI',
  'nightPI',

  'dayPC',
  'nightPC',

  'ch',

  'ge',

  'nv',

  'dayCV',
  'nightCV',

  'dayTR',
  'nightTR'
]

const iconByName = name => (
  <Content>
    <Icon name={name} width={50} height={50} />
    <Name>{name}</Name>
  </Content>
)

storiesOf(`${GROUPS.COMPONENTS}|Icon`, module)
  .addDecorator(story => (
    <Containerable>
      <Reset />
      {story()}
    </Containerable>
  ))
  .add('Outline', () => (
    <>
      {outlines.map(outline => (
        <Fragment key={outline}>{iconByName(outline)}</Fragment>
      ))}
    </>
  ))
  .add('Temperature', () => (
    <>
      {temperaturies.map(temperature => (
        <Fragment key={temperature}>{iconByName(temperature)}</Fragment>
      ))}
    </>
  ))
