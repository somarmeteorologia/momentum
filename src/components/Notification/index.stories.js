import React, { useState } from 'react'
import styled from 'styled-components'
import v4 from 'uuid/v4'
import { storiesOf } from '@storybook/react'

import { GROUPS } from '@helpers/hierarchySeparators'
import { Container } from '@helpers/components/Container'

import { Reset } from '@components/Reset'
import { Notification } from '@components/Notification'
import { Button } from '@components/Button'
import { Icon } from '@components/Icon'
import { Text } from '@components/Text'

const { keys } = Object

const generatePosition = state => {
  const [one, two, three] = [...keys(Notification.position)].slice(0, 3)

  const values = {
    primary: one,
    danger: two,
    warning: three
  }

  return values[state]
}

const Containerable = styled(Container)`
  flex-wrap: wrap;
  justify-content: space-evenly;
`

const Toggle = ({ children }) => {
  const [isOpen, setIsOpen] = useState(false)

  const toClose = () => setIsOpen(false)

  return (
    <>
      <Button onClick={() => setIsOpen(true)}>Open Notifications</Button>
      {children({ isOpen, toClose })}
    </>
  )
}

storiesOf(`${GROUPS.COMPONENTS}|Notification`, module)
  .addDecorator(story => (
    <Containerable>
      <Reset />
      {story()}
    </Containerable>
  ))
  .add('Default', () => (
    <>
      <Toggle>
        {({ isOpen, toClose }) => (
          <Notification
            icon={({ color }) => <Icon name="disabled" color={color} />}
            state={Notification.state.primary}
            toClose={toClose}
            isOpen={isOpen}
            title={({ color }) => (
              <Text.Heading size={Text.Heading.size.sixteen} color={color}>
                Lorem ipsum dolor sit amet
              </Text.Heading>
            )}
            description={({ color }) => (
              <Text size={Text.size.thirteen} color={color}>
                Excepteur sint occaecat cupidatat non proident, sunt in culpa
                qui officia deserunt mollit anim id est laborum.
              </Text>
            )}
          />
        )}
      </Toggle>
    </>
  ))
  .add('With duration (5s)', () => (
    <>
      <Toggle>
        {({ isOpen, toClose }) => (
          <Notification
            duration={5}
            icon={({ color }) => <Icon name="disabled" color={color} />}
            state={Notification.state.primary}
            toClose={toClose}
            isOpen={isOpen}
            title={({ color }) => (
              <Text.Heading size={Text.Heading.size.sixteen} color={color}>
                Lorem ipsum dolor sit amet
              </Text.Heading>
            )}
            description={({ color }) => (
              <Text size={Text.size.thirteen} color={color}>
                Excepteur sint occaecat cupidatat non proident, sunt in culpa
                qui officia deserunt mollit anim id est laborum.
              </Text>
            )}
          />
        )}
      </Toggle>
    </>
  ))
  .add('States', () => (
    <>
      <Toggle>
        {({ isOpen, toClose }) => (
          <>
            {keys(Notification.state).map(item => (
              <Notification
                key={v4()}
                state={item}
                position={generatePosition(item)}
                icon={({ color }) => <Icon name="disabled" color={color} />}
                toClose={toClose}
                isOpen={isOpen}
                title={({ color }) => (
                  <Text.Heading size={Text.Heading.size.sixteen} color={color}>
                    Lorem ipsum dolor sit amet
                  </Text.Heading>
                )}
                description={({ color }) => (
                  <Text size={Text.size.thirteen} color={color}>
                    Excepteur sint occaecat cupidatat non proident, sunt in
                    culpa qui officia deserunt mollit anim id est laborum.
                  </Text>
                )}
              />
            ))}
          </>
        )}
      </Toggle>
    </>
  ))
  .add('Positions', () => (
    <>
      <Toggle>
        {({ isOpen, toClose }) => (
          <>
            {keys(Notification.position).map(pos => (
              <>
                <Notification
                  key={v4()}
                  position={pos}
                  icon={({ color }) => <Icon name="disabled" color={color} />}
                  toClose={toClose}
                  isOpen={isOpen}
                  title={({ color }) => (
                    <Text.Heading
                      size={Text.Heading.size.sixteen}
                      color={color}
                    >
                      Lorem ipsum dolor sit amet
                    </Text.Heading>
                  )}
                  description={({ color }) => (
                    <Text size={Text.size.thirteen} color={color}>
                      Excepteur sint occaecat cupidatat non proident, sunt in
                      culpa qui officia deserunt mollit anim id est laborum.
                    </Text>
                  )}
                />
              </>
            ))}
          </>
        )}
      </Toggle>
    </>
  ))
