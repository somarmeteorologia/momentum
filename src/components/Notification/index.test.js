import React from 'react'
import { render, wait } from '@testing-library/react'

import { renderWithTheme, withTheme } from '../../test-utilities'

import { Theme } from '@components/Theme'
import { Icon } from '@components/Icon'
import { Text } from '@components/Text'
import { Notification } from '@components/Notification'

const { notification } = Theme.light

const Notificable = ({
  isOpen = false,
  state = Notification.state.primary
}) => (
  <Notification
    icon={({ color }) => <Icon name="disabled" color={color} />}
    state={state}
    toClose={() => {}}
    isOpen={isOpen}
    title={({ color }) => (
      <Text.Heading size={Text.Heading.size.sixteen} color={color}>
        Título
      </Text.Heading>
    )}
    description={({ color }) => (
      <Text size={Text.size.thirteen} color={color}>
        Descrição
      </Text>
    )}
  />
)

test('Not expect visual regression', () => {
  const { container } = renderWithTheme(<Notificable />)

  expect(container.firstChild).toMatchSnapshot()
})

const states = ['success', 'warning', 'primary', 'danger']

test('Expect correct state', () => {
  const { rerender, getByTestId } = render(
    withTheme(<Notificable isOpen={true} />)
  )

  states.forEach(state => {
    rerender(withTheme(<Notificable state={Notification.state[state]} />))

    expect(getByTestId('state')).toHaveStyleRule(
      'background-color',
      notification.state[state]
    )
  })
})

test('Expect not is open, after open', async () => {
  const { getByTestId, rerender } = render(
    withTheme(<Notificable isOpen={false} />)
  )

  expect(getByTestId('notification')).not.toBeVisible()

  rerender(withTheme(<Notificable isOpen={true} />))

  await wait(
    () => {
      expect(getByTestId('notification')).toBeVisible()
    },
    { interval: 500 }
  )
})
