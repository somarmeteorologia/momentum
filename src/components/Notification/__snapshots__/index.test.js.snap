// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not expect visual regression 1`] = `
.c6 {
  display: inline-block;
}

.c1 {
  display: inline-block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 0;
  cursor: pointer;
}

.c3 {
  display: inline-block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 0;
}

.c0 {
  width: 385px;
  height: 110px;
  padding: 16px;
  border-radius: 0.25rem;
  background-color: #111732;
  color: #fff;
  z-index: 9999;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 25px;
  right: 25px;
}

.c0 .close {
  position: absolute;
  top: 16px;
  right: 16px;
  cursor: pointer;
}

.c2 {
  width: 45px;
  min-width: 45px;
  height: 45px;
  min-height: 45px;
  border-radius: 0.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #133BEC;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 80px;
  padding-left: 16px;
  padding-right: 16px;
}

.c5 {
  margin-bottom: 5px;
  color: #FF0130;
}

.c7 {
  font-family: Inter,sans-serif;
  font-weight: 600;
  display: block;
  color: #fff;
  font-size: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 {
  font-family: Inter,sans-serif;
  font-weight: 500;
  display: block;
  color: #fff;
  font-size: 0.8125rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="c0"
  data-testid="notification"
  style="display: none; opacity: 0; transform: translateY(-150%) translateZ(0);"
>
  <div
    class="c1 close"
    height="14"
    width="14"
  >
    <svg
      fill="none"
      height="14"
      viewBox="0 0 21 20"
      width="14"
    >
      <path
        d="M20.75 1.99988L18.75 -0.00012207L10.75 7.99988L2.75 -0.00012207L0.75 1.99988L8.75 9.99988L0.75 17.9999L2.75 19.9999L10.75 11.9999L18.75 19.9999L20.75 17.9999L12.75 9.99988L20.75 1.99988Z"
        fill="#fff"
      />
    </svg>
  </div>
  <div
    class="c2"
    data-testid="state"
  >
    <div
      class="c3"
      height="20"
      width="20"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 32 32"
        width="20"
      >
        <mask
          fill="#fff"
          height="20"
          id="path-1-outside-1"
          maskUnits="userSpaceOnUse"
          width="20"
          x="0"
          y="0"
        >
          <rect
            fill="#fff"
            height="20"
            width="20"
          />
          <path
            clip-rule="evenodd"
            d="M16 1C7.72933 1 1 7.729 1 15.9997C1 24.271 7.72933 31 16 31C24.2707 31 31 24.2707 31 15.9997C31 7.729 24.271 1 16 1ZM2.06137 15.9997C2.06137 8.31368 8.31404 2.06068 16 2.06068C19.6527 2.06068 22.9797 3.47502 25.4677 5.78168L5.78237 25.4674C3.47504 22.979 2.06137 19.6527 2.06137 15.9997ZM6.53226 26.2177C9.02026 28.5243 12.3473 29.9387 15.9999 29.9387C23.6859 29.9387 29.9386 23.6857 29.9389 15.9993C29.9389 12.3467 28.5249 9.01999 26.2179 6.53166L6.53226 26.2177Z"
            fill-rule="evenodd"
          />
        </mask>
        <path
          clip-rule="evenodd"
          d="M16 1C7.72933 1 1 7.729 1 15.9997C1 24.271 7.72933 31 16 31C24.2707 31 31 24.2707 31 15.9997C31 7.729 24.271 1 16 1ZM2.06137 15.9997C2.06137 8.31368 8.31404 2.06068 16 2.06068C19.6527 2.06068 22.9797 3.47502 25.4677 5.78168L5.78237 25.4674C3.47504 22.979 2.06137 19.6527 2.06137 15.9997ZM6.53226 26.2177C9.02026 28.5243 12.3473 29.9387 15.9999 29.9387C23.6859 29.9387 29.9386 23.6857 29.9389 15.9993C29.9389 12.3467 28.5249 9.01999 26.2179 6.53166L6.53226 26.2177Z"
          fill="#fff"
          fill-rule="evenodd"
        />
        <path
          d="M25.4677 5.78168L25.6091 5.9231C25.6476 5.88467 25.6687 5.83225 25.6677 5.7779C25.6666 5.72356 25.6435 5.67197 25.6037 5.63502L25.4677 5.78168ZM5.78237 25.4674L5.63572 25.6033C5.67267 25.6432 5.72426 25.6663 5.7786 25.6673C5.83294 25.6683 5.88536 25.6472 5.92379 25.6088L5.78237 25.4674ZM6.53226 26.2177L6.39083 26.0762C6.3524 26.1147 6.33127 26.1671 6.33229 26.2214C6.33332 26.2758 6.35642 26.3274 6.39628 26.3643L6.53226 26.2177ZM29.9389 15.9993L30.1389 15.9993V15.9993H29.9389ZM26.2179 6.53166L26.3646 6.39569C26.3276 6.35583 26.276 6.33272 26.2217 6.3317C26.1674 6.33067 26.1149 6.35181 26.0765 6.39024L26.2179 6.53166ZM1.2 15.9997C1.2 7.83946 7.83978 1.2 16 1.2V0.8C7.61888 0.8 0.8 7.61854 0.8 15.9997H1.2ZM16 30.8C7.83979 30.8 1.2 24.1605 1.2 15.9997H0.8C0.8 24.3815 7.61888 31.2 16 31.2V30.8ZM30.8 15.9997C30.8 24.1602 24.1602 30.8 16 30.8V31.2C24.3811 31.2 31.2 24.3811 31.2 15.9997H30.8ZM16 1.2C24.1605 1.2 30.8 7.83946 30.8 15.9997H31.2C31.2 7.61854 24.3815 0.8 16 0.8V1.2ZM16 1.86068C8.20357 1.86068 1.86137 8.20323 1.86137 15.9997H2.26137C2.26137 8.42413 8.4245 2.26068 16 2.26068V1.86068ZM25.6037 5.63502C23.0804 3.29562 19.7054 1.86068 16 1.86068V2.26068C19.6 2.26068 22.879 3.65442 25.3317 5.92835L25.6037 5.63502ZM5.92379 25.6088L25.6091 5.9231L25.3263 5.64026L5.64095 25.3259L5.92379 25.6088ZM1.86137 15.9997C1.86137 19.7053 3.29563 23.0797 5.63572 25.6033L5.92903 25.3314C3.65445 22.8784 2.26137 19.6 2.26137 15.9997H1.86137ZM15.9999 29.7387C12.3999 29.7387 9.12092 28.3449 6.66823 26.071L6.39628 26.3643C8.91959 28.7037 12.2946 30.1387 15.9999 30.1387V29.7387ZM29.7389 15.9993C29.7386 23.5752 23.5755 29.7387 15.9999 29.7387V30.1387C23.7964 30.1387 30.1386 23.7961 30.1389 15.9993L29.7389 15.9993ZM26.0713 6.66764C28.3455 9.12066 29.7389 12.3993 29.7389 15.9993H30.1389C30.1389 12.294 28.7043 8.91933 26.3646 6.39569L26.0713 6.66764ZM6.67368 26.3591L26.3593 6.67308L26.0765 6.39024L6.39083 26.0762L6.67368 26.3591Z"
          fill="#fff"
          mask="url(#path-1-outside-1)"
        />
      </svg>
    </div>
  </div>
  <div
    class="c4"
  >
    <div
      class="c5"
    >
      <div
        class="c6"
        color="#fff"
        display="block"
      >
        <h6
          class="c7"
          color="#fff"
          display="block"
        >
          Título
        </h6>
      </div>
    </div>
    <div
      class="c6"
      color="#fff"
      display="block"
    >
      <span
        class="c8"
        color="#fff"
        display="block"
      >
        Descrição
      </span>
    </div>
  </div>
</div>
`;
